signer/rpc_and_client_test.go

```diff
@@ -80,73 +81,124 @@ func (s stubServer) Sign(ctx context.Context, sr *pb.SignatureRequest) (*pb.Sign
 }
 
 func (s stubServer) CheckHealth(ctx context.Context, v *pb.Void) (*pb.HealthStatus, error) {
-	res, err := s.stubHealthFunc()
-	if err != nil {
-		return nil, err
-	}
-	return &pb.HealthStatus{Status: res}, nil
+	return &pb.HealthStatus{}, nil
 }
 
-func getStubbedHealthServer(healthFunc func() (map[string]string, error)) *grpc.Server {
-	s := stubServer{stubHealthFunc: healthFunc}
+func getStubbedHealthServer(hs *health.Server) *grpc.Server {
+	s := stubServer{healthServer: hs}
 	gServer := grpc.NewServer()
 	pb.RegisterKeyManagementServer(gServer, s)
 	pb.RegisterSignerServer(gServer, s)
+
+	if s.healthServer != nil {
+		healthpb.RegisterHealthServer(gServer, s.healthServer)
+	}
+
 	return gServer
 }
 
 // CheckHealth does not succeed if the KM server is unhealthy
 func TestHealthCheckKMUnhealthy(t *testing.T) {
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return map[string]string{"health": "not good"}, nil
-	})
+	hs := health.NewServer()
+	hs.SetServingStatus(notary.HealthCheckKeyManagement, healthpb.HealthCheckResponse_NOT_SERVING)
```

Comment:
I'm really happy with this test and the bunch of others (`TestHealthCheckKMError`, `TestHealthCheckKMTimeout`, etc.) for `HealthCheckKeyManagement` but I was wondering if we could also repeat the same tests for `HealthCheckSigner`?

We can factor out the existing tests to take a service name and check against that individual service being degraded.  Also, I was wondering if we should check against both the degraded service as well as the overall signer service (`HealthCheckOverall`)?